arkouda/alignment.py

```diff
@@ -223,36 +275,18 @@ def lookup(keys, values, arguments, fillvalue=-1, keys_from_unique=False):
     (array(['twenty', 'twenty', 'twenty']),
     array(['four', 'one', 'two']))
     """
-    if not keys_from_unique:
-        keyg = GroupBy(keys)
-        if keyg.length != keyg.ngroups:
-            raise NonUniqueError("Function keys must be unique.")
-        keys = keyg.unique_keys
-        values = values[keyg.permutation]
     if isinstance(values, Categorical):
         codes = lookup(keys, values.codes, arguments, fillvalue=values._NAcode)
         return Categorical.from_codes(codes, values.categories, NAvalue=values.NAvalue)
-    # Condense down to unique arguments to query
-    g = GroupBy(arguments)
-    # scattermask = Args that exist in table
-    # gathermask = Elements of keys that are being asked for
-    try:
-        scattermask, gathermask = in1dmulti(g.unique_keys, keys, assume_unique=True, symmetric=True)
-    except NonUniqueError:
-        raise NonUniqueError("Function keys must be unique.")
-    # uvals = Retrieved values corresponding to unique args being queried
-    if g.nkeys == 1:
-        uvals = zeros(g.unique_keys.size, dtype=values.dtype)
-    else:
-        uvals = zeros(g.unique_keys[0].size, dtype=values.dtype)
-    # Use default value for arguments not present in keys
-    if fillvalue is not None:
-        uvals.fill(fillvalue)
-    # Set values for arguments present in keys
-    # Set them to the values corresponding to elements queried
-    uvals[scattermask] = values[gathermask]
-    # Broadcast return values back to non-unique arguments
-    return g.broadcast(uvals, permute=True)
+    # Find arguments in keys array
+    idx = find(arguments, keys)
+    # Initialize return values with fillvalue for missing values
+    retvals = ak.zeros(idx.size, dtype=values.dtype)
+    retvals.fill(fillvalue)
```

Comment:
I think at least part of the problem is the `ak.` shouldn't be here. Also you can do this in one line using `full`
```python
retvals = full(idx.size, fillvalue, dtype=values.dtype)
```

Reply:
Ah, thanks!